musit-service/src/main/scala/no/uio/musit/ws/ProxiedRequest.scala

```diff
@@ -0,0 +1,35 @@
+package no.uio.musit.ws
+
+import no.uio.musit.ws.ProxiedRequest._
+import play.api.Configuration
+import play.api.libs.ws.{DefaultWSProxyServer, WSProxyServer, WSRequest}
+
+class ProxiedRequest(req: WSRequest, config: Configuration) {
+
+  val proxy: Option[WSProxyServer] = for {
+    host <- config.getString(ProxyHost)
+    port <- config.getInt(ProxyPort)
+    user     = config.getString(ProxyUser)
+    password = config.getString(ProxyPassword)
+  } yield
+    DefaultWSProxyServer(host = host, port = port, principal = user, password = password)
+
+  implicit def viaProxy(): WSRequest = proxy.map(req.withProxyServer).getOrElse(req)
+
+}
+
+object ProxiedRequest {
+
+  val ProxyHost     = "ws.proxy.host"
+  val ProxyPort     = "ws.proxy.port"
+  val ProxyUser     = "ws.proxy.user"
+  val ProxyPassword = "ws.proxy.password"
+
+}
+
+trait ViaProxy {
+
+  implicit def viaProxy(req: WSRequest)(implicit config: Configuration): ProxiedRequest =
+    new ProxiedRequest(req, config)
+
+}
```

Comment:
It's a good idea to also add an `object` of type `ViaProxy` to have a singleton available for direct import of the implicit converter.

That way you don't need to have strange extensions on the `class` definitions in cases where you need a `WSClient` with the `viaProxy` function available in implicit scope. Instead it will be implicitly available from the imported object.

See e.g. `no.uio.musit.formatters.WithDateTimeFormatters` and `no.uio.musit.formatters.DateTimeFormatters`